exasol/toolbox/tools/security.py

```diff
@@ -0,0 +1,266 @@
+"""This module contains security related CLI tools and code"""
+import json
+import re
+import subprocess
+import sys
+from dataclasses import (
+    asdict,
+    dataclass,
+)
+from enum import Enum
+from functools import partial
+from inspect import cleandoc
+from typing import (
+    Generator,
+    Iterable,
+    Tuple,
+)
+
+import typer
+
+stdout = print
+stderr = partial(print, file=sys.stderr)
+
+
+# Note: In the long term we may want to adapt the official CVE json schema
+# https://github.com/CVEProject/cve-schema/blob/master/schema/v5.0/CVE_JSON_5.0_schema.json
+@dataclass(frozen=True)
+class Issue:
+    cve: str
+    cwe: str
+    description: str
+    coordinates: str
+    references: tuple
+
+
+def _issues(input) -> Generator[Issue, None, None]:
+    lines = (l for l in input if l.strip() != "")
+    for line in lines:
+        obj = json.loads(line)
+        yield Issue(**obj)
+
+
+def _issues_as_json_str(issues):
+    for issue in issues:
+        issue = asdict(issue)  # type: ignore
+        yield json.dumps(issue)
+
+
+def gh_security_issues() -> Generator[Tuple[str, str], None, None]:
+    """
+    Yields issue-id, cve-id pairs for all (closed, open) issues associated with CVEs
+
+    Return:
+        A generator which yield tuples of (id,title).
+
+    Raises:
+        subprocess.CalledProcessError: If the underlying command fails.
+    """
+    command = [
+        "gh",
+        "issue",
+        "list",
+        "--label",
+        "security",
+        "--search",
+        "CVE",
+        "--json",
+        "id,title",
+        "--limit",
+        "1000",
+        "--state",
+        "all",
+    ]
+    try:
+        result = subprocess.run(command, check=True, capture_output=True)
+    except FileNotFoundError as ex:
+        msg = "Command 'gh' not found. Please make sure you have installed the github cli."
+        raise FileNotFoundError(msg) from ex
+    except subprocess.CalledProcessError as ex:
+        stderr(f"{ex}")
+        raise ex
+
+    cve_pattern = re.compile(r"CVE-\d{4}-\d{4,7}")
+    issues = json.loads(result.stdout.decode("utf-8"))
+    issues = (
+        (issue["id"], cve_pattern.search(issue["title"]).group())  # type: ignore
+        for issue in issues
+        if cve_pattern.search(issue["title"])
```

Comment:
this is search is quite fuzzy, you could get tickets that mention a CVE, but are not actually there for fixing them

Reply:
As discussed, potentially we will change this in the future.